conwayste/src/client.rs

```
@@ -904,20 +934,53 @@ impl MainState {
                     .scale(Vector2::new(rect.w, rect.h))
                     .color(color);
 
-                spritebatch.add(p);
+                main_spritebatch.add(p);
             }
         });
 
+        // TODO: truncate if outside of writable region
+        // TODO: move to new function
+        if let Some((ref grid, width, height)) = self.insert_mode {
+            if self.uni_draw_params.player_id < 0 {
+                return Err(format!("Unexpected player ID {}", self.uni_draw_params.player_id).into());
+            }
+            let player_cell_state = CellState::Alive(Some(self.uni_draw_params.player_id as usize));
+            let mut color = self.color_settings.get_color(Some(player_cell_state));
+            color.a = 0.5;  // semi-transparent since this is an overlay
+            if let Some(cursor_cell) = viewport.game_coords_from_window(self.inputs.mouse_info.position) {
+                let (cursor_col, cursor_row) = (cursor_cell.col, cursor_cell.row);
+                grid.each_set(|grid_col, grid_row| {
+                    let col = (grid_col + cursor_col) as isize - width as isize/2;
+                    let row = (grid_row + cursor_row) as isize - height as isize/2;
+                    if col < 0 || row < 0 {
+                        // out of range
+                        return;
```

Silently return? Why no error?